librz/core/cdwarf.c

```diff
@@ -3,75 +3,109 @@
 
 #include <rz_core.h>
 
-RZ_API void rz_core_bin_dwarf_print_abbrev_section(const RzBinDwarfDebugAbbrev *da) {
-	size_t i, j;
-	if (!da) {
-		return;
+#define my_printf(...) rz_strbuf_appendf(sb, __VA_ARGS__)
+#define my_print(x)    rz_strbuf_append(sb, (x))
+#define my_print_init \
+	RzStrBuf *sb = rz_strbuf_new(NULL); \
+	if (!sb) { \
+		return NULL; \
 	}
-	for (i = 0; i < da->count; i++) {
-		rz_cons_printf("   %-4" PFMT64d " ", da->decls[i].code);
-		const char *tagname = rz_bin_dwarf_get_tag_name(da->decls[i].tag);
-		if (tagname) {
-			rz_cons_printf("  %-25s ", tagname);
+#define my_print_get rz_strbuf_drain(sb)
+
+RZ_API char *rz_core_bin_dwarf_abbrev_decl_to_string(RzBinDwarfAbbrevDecl *decl) {
```

Comment:
As these allocate new strings, please add ownership to the return values

Reply:
```suggestion
RZ_API RZ_OWN char *rz_core_bin_dwarf_abbrev_decl_to_string(RZ_BORROW RzBinDwarfAbbrevDecl *decl) {
```